src/opera_disp_tms/generate_frame_tile.py

```diff
@@ -0,0 +1,332 @@
+    feature = ogr.Feature(layer.GetLayerDefn())
+    feature.SetGeometry(ogr_polygon)
+    layer.CreateFeature(feature)
+
+    # Rasterize the polygon onto the output raster
+    gdal.RasterizeLayer(tmp_ds, [1], layer, burn_values=[frame.frame_id])
+
+    # Close and flush the dataset to disk
+    tmp_band.FlushCache()
+    tmp_ds = None
+
+    # Reopen the dataset and copy the data to the tile
+    frame_ds = gdal.Open(str(tmp_tiff))
+    frame_array = frame_ds.GetRasterBand(1).ReadAsArray()
+
+    tile_band = tile_ds.GetRasterBand(1)
+    tile_array = tile_band.ReadAsArray()
+    tile_array[frame_array != 0] = frame_array[frame_array != 0]
+    tile_band.WriteArray(tile_array)
+
+    # Clean up
+    frame_ds.FlushCache()
+    frame_ds = None
+    tile_band.FlushCache()
+    tile_ds = None
+    tmp_tiff.unlink()
+
+
+def get_granule_reference_info_s3(s3_granule_path: str) -> Tuple:
+    """Get the reference information for an OPERA DISP granule stored in S3
+
+    Args:
+        s3_granule_path: The S3 URI to the granule
+
+    Returns:
+        A tuple containing the reference point in array coordinates, geographic coordinates, and EPSG code
+    """
+    io_params = {
+        's3fs_params': {
+            'default_cache_type': 'blockcache',
+            'default_block_size': 8 * 1024 * 1024,
+        },
+        'h5py_params': {
+            'driver_kwds': {
+                'page_buf_size': 32 * 1024 * 1024,
+                'rdcc_nbytes': 8 * 1024 * 1024,
+            }
+        },
+    }
+    creds = get_credentials()
+    fs = s3fs.S3FileSystem(
+        key=creds['accessKeyId'],
+        secret=creds['secretAccessKey'],
+        token=creds['sessionToken'],
+        **io_params['s3fs_params'],
+    )
+    with fs.open(s3_granule_path, 'rb') as f:
+        with h5py.File(f, 'r', **io_params['h5py_params']['driver_kwds']) as hdf:
+            ref_point_array, ref_point_geo, epsg = read_reference_info(hdf)
+    return ref_point_array, ref_point_geo, epsg
+
+
+def read_reference_info(h5_fobj: h5py.File) -> Tuple:
+    """Find the reference information for an OPERA DISP granule
+
+    Args:
+        h5_fobj: An open HDF5 file object for an OPERA DISP granule
+
+    Returns:
+        A tuple containing the reference point in array coordinates, geographic coordinates, and EPSG code
+    """
+    ref_point = h5_fobj['corrections']['reference_point'].attrs
+    ref_col = int(ref_point['cols'][0])
+    ref_row = int(ref_point['rows'][0])
+    ref_lat = float(ref_point['latitudes'][0])
+    ref_lon = float(ref_point['longitudes'][0])
+
+    srs = osr.SpatialReference()
+    srs.ImportFromWkt(h5_fobj['spatial_ref'].attrs['crs_wkt'])
+    epsg = int(srs.GetAuthorityCode(None))
+
+    ref_point_geo = (ref_lon, ref_lat)
+    ref_point_array = (ref_col, ref_row)
+    return ref_point_array, ref_point_geo, epsg
+
+
+def add_metadata_to_tile(tile_path: Path, ascending=True) -> None:
```

Comment:
```suggestion
def add_metadata_to_tile(tile_path: Path) -> None:
```
ascending is never called in this func